src/utils/hsdis/binutils/hsdis-binutils.c

```diff
@@ -336,8 +336,8 @@ static void setup_app_data(struct hsdis_app_data* app_data,
                                  app_data->printf_stream,
                                  app_data->printf_callback,
                                  native_bfd,
-                                 /* On PowerPC we get warnings, if we pass empty options */
-                                 (caller_options == NULL) ? NULL : app_data->insn_options);
+                                 /* On some archs we get warnings, if we pass empty options */
+                                 (app_data->insn_options[0] == '\0') ? NULL : app_data->insn_options);
```

Comment:
What if `caller_options == NULL`?

Reply:
The parse_caller_options handles the NULL case, so I forgot about the early bailout.
```
  for (p = caller_options; p != NULL; ) {
  }
  *iop = '\0';

```
Sorry.

Reply:
Just remove the if, and always call parse_caller_options ?

Reply:
Sorry again, long day:
```
  struct hsdis_app_data app_data;
  memset(&app_data, 0, sizeof(app_data));
```
I knew I saw that somewhere.
So if we don't parse at all (caller options NULL) it should be empty string in insn_options.

That is the way we handle caller option NULL, no ?

Reply:
Oh, I see.